models/DataObject/AbstractObject.php

```diff
@@ -411,9 +411,11 @@ protected static function typeMatch(AbstractObject $object)
      *
      * @return self[]
      */
-    public function getChildren($objectTypes = [self::OBJECT_TYPE_OBJECT, self::OBJECT_TYPE_FOLDER], $unpublished = false)
+    public function getChildren(array $objectTypes = [self::OBJECT_TYPE_OBJECT, self::OBJECT_TYPE_FOLDER], $unpublished = false)
     {
-        if ($this->o_children === null || $this->lastGetChildrenObjectTypes != $objectTypes) {
+        $cacheKey = implode('_', $objectTypes) . (!empty($objectTypes) ?'_' : '') . (int) $unpublished;
```

Comment:
We should move that to a `protected` function so that we can reuse it in `hasChildren()` `getSiblings()` ... 
Maybe something generic like: 
```php
$this->getListingCacheKey(func_get_args()); 
```

Reply:
We should also cast to `string` instead of `int` as it get's converted to `string` anyway.